src/scripts/main.js

```diff
@@ -1 +1,80 @@
 'use strict';
+
+async function wait(ms) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+const firstPromise = new Promise(async (resolve, reject) => {
+  let isTrue = false;
+
+  document.addEventListener('click', () => {
+    isTrue = true;
+    resolve('First promise was resolved');
+  });
+
+  await wait(3000);
+
+  if (!isTrue) {
+    reject(new Error('First promise was rejected'));
+  }
+});
+
+const secondPromise = new Promise((resolve, reject) => {
+  let isTrue = false;
+
+  document.addEventListener('click', () => {
+    resolve('Second promise was resolved');
+    isTrue = true;
+  });
+
+  if (isTrue !== true) {
+    document.addEventListener('contextmenu', () => {
+      resolve('Second promise was resolved');
+    });
+  }
+});
```

Comment:
Bug: The secondPromise will always be resolved, even if the user doesn't click or right-click. This is because the check for `isTrue` is done immediately after the event listener is set up, and at that point, `isTrue` is always `false`. To fix this, you could use `setTimeout` to delay the check for `isTrue` and the setup of the second event listener. This way, the promise will be resolved only if the user clicks or right-clicks within a certain time after the page load.